### DISCUSSION TITLE
Login with Username instead of Email (Extend authentication parameters)

### DISCUSSION THREAD

**Discussion by TomDoFuture** [payloadcms/payload#1196] [Feature Requests & Ideas]
Currently one needs Email and Password to login as a user.

Would it be possible if I can create a text field called Username, and then login with the username instead of the email? So you can login with both.

**Comment by jmikrut** [comment]
Hey @TomDoFuture — yes, you can build a custom auth strategy to accomplish this if you'd like. As of now, Payload only supports email-based local authentication but you can build your own using Passport strategies!

**Comment by leesei** [comment]
The [docs on strategy](https://payloadcms.com/docs/authentication/config#strategies) lacks sample code.
Where can I found an example usage of this feature?

Specifically I would like the auth strategyto only apply to a custom collection, they do not need access to the Admin UI. The authentication status can be used for authorization in CRUD on another collection (together with the existing User collection).

I've found two auth plugin but they are a bit too complicated for my use case.
[thgh/payload-plugin-oauth: Add oAuth sign in to your Payload CMS site](https://github.com/thgh/payload-plugin-oauth)
[finkinfridom/payload-auth0-plugin: Extends payloadcms with Auth0 integration](https://github.com/finkinfridom/payload-auth0-plugin)

Any pointer to how to implementation a simple auth plugin? I can contribute it back to the community.

**Reply by DanRibbens** [reply]
@leesei, I don't think it is a custom strategy you're after. It sounds to me that you want to expand on the default access control and add another auth enabled collection, separate from your users. You can already do this so I don't think there is any need for a contribution to core. Payload's built in auth strategies are going to either use email and passwords or you can use `auth.useAPIKey` and optionally, disable `localeAuthStrategy` to turn off password auth for that collection. API keys are better for programmatic access.

All you need to do is add an additional auth enabled collection, using the settings I described as they fit your needs and then write your access control functions to be conditional around the `user.collection` to limit access in the custom way you described. That way a user authed as `collection: 'users'` can have different access from another authed user logged in as `collection: 'newbs'` let's say.

Does that fit your use-case?

**Reply by DanRibbens** [reply]
Also you can choose which auth collection is able to access the admin UI in your Payload config with the `admin.users: 'users'` where "users" is the slug.

Feel free to open another discussion if you get stuck, replies can be hard to track and end up unanswered sometimes.
Good luck!

**Reply by leesei** [reply]
Thanks for the reply.
I'll create a new thread.
